tests/ui/pattern/deref-patterns/ice-adjust-mode-unimplemented-for-constblock.rs

```diff
@@ -0,0 +1,17 @@
+// Regression test for #148138
+// Ensure that const blocks in patterns are rejected with a proper error
+// instead of causing an ICE in the type checker.
+//
+// The vec! macro expansion bypasses parser validation for const blocks in patterns,
+// so this must be caught during AST lowering.
+
+#![feature(deref_patterns)]
+//~^ WARN the feature `deref_patterns` is incomplete
```

Comment:
If you use `#![expect(...)]`, the WARN will not occur.

```suggestion
#![feature(deref_patterns)]
#![expect(incomplete_features)]
```